src/components/TodoList.js

```diff
@@ -1,36 +1,53 @@
 import React, { useEffect } from 'react'
 import ListItem from './ListItem'
-import { connect } from 'react-redux'
-import { getTodos } from '../store/todos'
+
+/*
+  TodoList component
+  - Function: TodoList is a presentational component that displays a list of todos as
+              the user adds them
+    Props:    It receives the following props from the Homepage component:
+              - todos - list of todos held in state (Redux store)
+              - editTodo - edits a single todo list item
+              - deleteTodo - deletes a todo from the list
+              - refreshTodos - updates to the current list of todos after a change been made
+              - getTodo - fetches a single todo item
+    Children: ListItem
+                - ListItem will receive as props from TodoList the following:
+                    - all of the properties of each individual todo
+                          - id,
+                          - task,
+                          - completed
+                    - editTodo - edits a single todo item
+                    - deleteTodo - deletes a single todo item
+                    - getTodo - fetches a single todo from the Redux store. This item will also
+                                correspond to the react router route ('/todos/:id) with id being
+                                the id of the todo item
+
+*/
 
 const TodoList = (props) => {
+  const { todos, editTodo, deleteTodo, getTodo, refreshTodos } = props
   useEffect(() => {
-    async function getData () {
-      await props.getTodos()
+    async function fetchTodos () {
+      await refreshTodos()
     }
-    getData()
-  })
-
-  const { todos } = props
+    fetchTodos()
+  }, todos)
```

Comment:
The second arg passed to `useEffect` is a list of dependencies to trigger the effect to be invoked. Unless I'm misunderstanding, this could cause a loop where `fetchTodos` triggers update to redux, which then updates `todos` which then triggers the effect to call `fetchTodos` and so on.

Building off my first comment, perhaps this hook belongs in `Homepage` with an empty array passed as the second argument. That will make the hook only run once, on component mount. Then subsequent refreshes can be triggered by user interaction.

Just a thought. Let me know if I am misunderstanding.

Reply:
I thought the second argument was a dependency array, which re-renders the component every time that there is a change in that variable.  An empty array equates to ComponentDidMount and would only run once.  useEffect with a dependency array is more like ComponentDidUpdate and re-renders every time there is a component change except it re-renders with changes to the values inside the dependency array.

Reply:
You're correct. The `useEffect` will fire when _any_ of the values in the array change. And since the `fetchTodos()` will presumably update the `todos` to fresh objects, and then those are passed as dependencies to trigger reinvoking the hook, you get a situation like `useEffect(fetchTodos()) -> new todos -> useEffect(fetchTodos()) -> new todos -> useEffect(fetchTodos())...`.

My understanding is that you would only want to invoke the hook once, to initialize the todos in redux, then subsequent updates to the store would propagate down through `todos`.

What do you think? If you have a design in mind that i'm not understanding, then I defer to you and this can wait until your full flow is in place. Let me know.